lua/chatcommands/coinflip.lua

```diff
@@ -0,0 +1,89 @@
+CoinFlips = CoinFlips or {}
+RegisterChatCommand({'coin','coinflip'},function(ply,arg)
+	if arg:lower() == "accept" then checkCoinFlipRequest(ply) return end
+	local t = string.Explode(" ",arg)
+	local p = tonumber(table.remove(t))
+	if p == nil then
+		ply:ChatPrint("[orange]!coinflip player points")
+	else
+		local to,c = PlyCount(string.Implode(" ",t))
+		if c == 1 then
+			if ply == to then
+				ply:ChatPrint("[red]You can't coinflip yourself!")
+			elseif p >= 1000 then -- minimum 1000 point coinflip
+				initCointFlip(ply,to,p)
+			else
+				ply:ChatPrint("[red]A coin flip must be a minimum of 1000 points. No pussy bets!")
+			end
+		else
+			ply:ChatPrint("[red]Player "..string.Implode(" ",t).." not found")
+		end
+	end
+end,{global=true,throttle=true})
+
+function initCointFlip(ply,target,amount)
+	if ply:PS_HasPoints(amount) and target:PS_HasPoints(amount) and CoinFlips[ply:SteamID()] == nil then
+		-- Both players have enough points AND an existing coinflip doesn't exist.
+		ply:ChatPrint("[orange]"..to:Nick().." is receiving your coinflip request.")
+		CoinFlips[ply:SteamID()] = {target:SteamID(), amount}
+		timer.Simple(120, CoinFlipRequestTimeout, ply, target) -- 2 minutes for the timeout.
+	elseif CoinFlips[ply:SteamID()] ~= nil then
+		ply:ChatPrint("[red]What the hell r yeh doing bruv? yeh already hae a filp goin!")
+	elseif not ply:PS_HasPoints(amount) then
+		ply:ChatPrint("[red]What the hell r yeh doing bruv? yeh don't hae dooze points!")
+	elseif not target:PS_HasPoints(amount) then
+		-- Should a player be told another player doesn't have enough points?
+		ply:ChatPrint("[red]y'r bruv doesn't hae dooze points to coinflip with yeh!")
+	end
+end
+
+local function CoinFlipRequestTimeout(ply, target)
+	table.remove(CoinFlips, ply:SteamID())
+	ply:ChatPrint("[fbc]The other bruv don't want to play. Try again lates.")
+	target:ChatPrint("[fbc]Missed out on a coinflip from " .. ply:Nick() .. ".")
+end
+
+function checkCoinFlipRequest(target)
+	-- There is probably a better way to do this. I have always found this kind of coding nasty once you get a sip of Java 8.
+	coinflipFound = false
+	for i,j in pairs(CoinFlips) do
```

Comment:
So if multiple people all give one guy requests, it will accept all of them? It should only do the most recently sent one

Reply:
Should I instead prevent multiple requests to the same person?